deploy/setup/setup.sh

```diff
@@ -0,0 +1,199 @@
+LOCALHOST="${LOCALHOST:-false}"
+SKIP_CLONE="${SKIP_CLONE:-false}"
+SKIP_DEPENDENCY_INSTALLS="${SKIP_DEPENDENCY_INSTALLS:-false}"
+SKIP_LAUNCH="${SKIP_LAUNCH:-false}"
+REPO_URL="${REPO_URL:-https://github.com/wmde/wikibase-release-pipeline.git}"
+USE_WEB="${USE_WEB:-false}"
+VERBOSE="${VERBOSE:-false}"
+WBS_DIR="${WBS_DIR:-/opt/wbs}"
+DEPLOY_DIR="${DEPLOY_DIR:-$WBS_DIR/wikibase-release-pipeline/deploy}"
+SETUP_DIR="${SETUP_DIR:-$DEPLOY_DIR/setup}"
+SCRIPTS_DIR="$SETUP_DIR/scripts"
+LOG_PATH="${LOG_PATH:-/tmp/wbs-deploy-setup.log}"
+
+# -- Setup logging --
+# NOTE: This logging functionality should be kept in sync
+# with the scripts/_logging.sh library used in the other setup
+# scripts. That library can't otherwise be directly used here as
+# this script is designed to be initiated before the repo is
+# cloned or available.
+mkdir -p "$(dirname "$LOG_PATH")"
+touch "$LOG_PATH" 2>/dev/null || true
+
+log() {
+  if $VERBOSE; then
+    echo "$@"
+  fi
+}
+
+log_cmd() {
+  if $VERBOSE; then
+    bash -c "$@"
+  else
+    bash -c "$@" &> /dev/null
+  fi
+  return $?
+}
+
+install_git() {
+  if command -v git &>/dev/null; then
+    log "Git already installed"
+    return
+  fi
+
+  echo "Installing Git..."
+  if command -v apt-get &>/dev/null; then
+    log_cmd "apt-get update && apt-get install -y git"
+  elif command -v dnf &>/dev/null; then
+    log_cmd "dnf install -y git"
+  elif command -v yum &>/dev/null; then
+    log_cmd "yum install -y git"
+  elif command -v apk &>/dev/null; then
+    log_cmd "apk add git"
+  elif command -v pacman &>/dev/null; then
+    log_cmd "pacman -Sy --noconfirm git"
+  else
+    echo "\u26a0\ufe0f Unsupported package manager. Please install Git manually."
+    exit 1
+  fi
+}
+
+install_docker() {
+  echo "Installing Docker..."
+
+  # shellcheck disable=SC1091
+  . /etc/os-release
+  case "$ID" in
+    ubuntu|debian)
+      log "Detected OS: $ID"
+
+      log_cmd "apt-get update"
+      log_cmd "apt-get install -y ca-certificates curl gnupg lsb-release"
+
+      log_cmd "mkdir -p /etc/apt/keyrings"
+      curl -fsSL https://download.docker.com/linux/$ID/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+
+      echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] \
+        https://download.docker.com/linux/$ID $(lsb_release -cs) stable" | \
+        tee /etc/apt/sources.list.d/docker.list > /dev/null
+
+      log_cmd "apt-get update"
+      log_cmd "apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin"
+      ;;
+
+    fedora)
+      log "Detected OS: Fedora"
+
+      log_cmd "dnf install -y dnf-plugins-core"
+      log_cmd "dnf config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo"
+      log_cmd "dnf install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin"
+      log_cmd "systemctl enable --now docker"
+      ;;
+
+    centos)
```

Comment:
My initial research revealed no centos docker/compose packages, only podman. Maybe we need to stick with the additional repository on centos. :(

Reply:
By the additional repo, do you mean what the convenience script sets up? If so, then I agree. I would prefer to resort to the Docker provided convenance script over the two alternatives I see: 1) not supporting the dist+version, or, 2) creating and maintaining our own dist+version specific steps. Given that, the question which remains for me is just how much of a notice/warning/fuss is appropriate and helpful to give to the user in these cases.

Reply:
Like the approach.

> how much of a notice/warning/fuss is appropriate and helpful to give to the user in these cases

Hm, maybe a note that we are falling back to https://get.docker.com is nice...